File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int maxScore = Integer.MIN_VALUE;
        int totalOnes = 0;
        int n = s.length();
        
        for(char c: s.toCharArray()){
            if(c=='1'){
                totalOnes++;
            }
        }

        int leftZeros = 0;
        int rightOnes = totalOnes;

        for(int i = 0; i<n-1; i++){
            if(s.charAt(i)=='0'){
                leftZeros++;
            }else{
                rightOnes--;
            }

            int currentScore = leftZeros + rightOnes;
            maxScore = Math.max(currentScore, maxScore);
        }

        return maxScore;
    }
}

File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        int countZeroes = 0;
        int countOnes = 0;
        int score;
        int maxScore = INT_MIN;

        for(int i = 0; i<s.size(); i++){
            if(s[i]=='1'){
                countOnes++;
            }
        }

        for(int i = 0; i<s.size()-1; i++){
            if(s[i]=='0'){
                countZeroes++;
            }else{
                countOnes--;
            }
            score = countZeroes + countOnes;

            maxScore = max(score, maxScore);

        }

        return maxScore;
    }
};